deepfence_frontend/apps/dashboard/src/features/vulnerabilities/pages/RuntimeBom.tsx

```diff
@@ -18,132 +14,34 @@ import {
-      },
-      in_field_filter: null,
-      window: {
-        offset: 0,
-        size: 0,
-      },
-    },
-    scan_filters: {
-      filters: {
-        contains_filter: {
-          filter_in: {
-            status: ['COMPLETE'],
-          },
-        },
-        match_filter: {
-          filter_in: {},
-        },
-        order_filter: {
-          order_fields: [
-            {
-              field_name: 'updated_at',
-              descending: true,
-            },
-          ],
-        },
-        compare_filter: null,
-      },
-      in_field_filter: null,
-      window: {
-        offset: 0,
-        size: 0,
-      },
-    },
-    window: {
-      offset: page * PAGE_SIZE,
-      size: PAGE_SIZE,
-    },
-  };
-  const order = getOrderFromSearchParams(searchParams);
-  if (order) {
-    requestFilters.node_filters.filters.order_filter.order_fields = [
-      {
-        field_name: order.sortBy,
-        descending: order.descending,
-      },
-    ];
-  }
-  const searchVulnerabilityScanApi = apiWrapper({
-    fn: getSearchApiClient().searchVulnerabilityScan,
-  });
-  const searchVulnerabilityScanResponse = await searchVulnerabilityScanApi({
-    searchSearchScanReq: requestFilters,
-  });
-  if (!searchVulnerabilityScanResponse.ok) {
-    throw new Error('unknown response');
-  }
-
-  const searchVulnerabilityScanCountApi = apiWrapper({
-    fn: getSearchApiClient().searchVulnerabilityScanCount,
-  });
-  const searchVulnerabilityScanCountResponse = await searchVulnerabilityScanCountApi({
-    searchSearchScanReq: {
-      ...requestFilters,
-      window: {
-        ...requestFilters.window,
-        size: 10 * requestFilters.window.size,
-      },
-    },
+const RuntimeBom = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { data } = useSuspenseQuery({
+    ...queries.vulnerability.scanList({
+      pageSize: PAGE_SIZE,
+      clusters: searchParams.getAll('clusters'),
+      containers: searchParams.getAll('containers'),
+      hosts: searchParams.getAll('hosts'),
+      images: searchParams.getAll('containerImages'),
+      languages: searchParams.getAll('languages'),
+      page: parseInt(searchParams.get('page') ?? '0', 10),
+      order: getOrderFromSearchParams(searchParams),
+      status: ['COMPLETE'],
+    }),
+    keepPreviousData: true,
   });
-  if (!searchVulnerabilityScanCountResponse.ok) {
-    throw searchVulnerabilityScanCountResponse.error;
-  }
-
-  return {
-    scans: searchVulnerabilityScanResponse.value,
-    currentPage: page,
-    totalRows: page * PAGE_SIZE + searchVulnerabilityScanCountResponse.value.count,
-  };
-}
-
-type LoaderData = {
-  scans: Awaited<ReturnType<typeof getScans>>;
-};
-
-const loader = async ({
-  request,
-}: LoaderFunctionArgs): Promise<TypedDeferredData<LoaderData>> => {
-  const searchParams = new URL(request.url).searchParams;
-  return typedDefer({
-    scans: getScans(searchParams),
+  const isFetching = useIsFetching({
```

Comment:
This doesn't work, this component should be wrapped in <Suspense> in parent. Just try to navigate from vulnerability dashboard to this page, it will throw an error.